### PR TITLE
Add encode/decode to AnySigner

### LINKED ISSUE #946: [AnySigner] Encode proto transaction message to blockchain raw bytes
```kotlin
val submittable: ByteArray = AnySigner.encode(signingInput, coinType)
```

### LINKED ISSUE #947: [AnySigner] Decode blockchain raw bytes to representable format
```kotlin
val decoded = AnySigner.decode(rawBytes, coinType)
// for ethereum, it's almost the same json from eth_getTransactionByHash
```

### PR DESCRIPTION
**Pull request by hewigovens**
Fixes #947
Fixes #946 

Only Ethereum is implemented now: 
- encode: signing input -> RLP encoded raw tx (can be sent via RPC)
- decode: RLP raw tx -> json data (almost same as json returned from `eth_getTransactionByHash`
- nested RLP list is not supported yet

### CODE REVIEW

File: include/TrustWalletCore/TWAnySigner.h [hunk 1]
```diff
@@ -23,6 +23,12 @@ extern TWString *_Nonnull TWAnySignerSignJSON(TWString *_Nonnull json, TWData *_
 
 extern bool TWAnySignerSupportsJSON(enum TWCoinType coin);
 
+/// Encodes a broadcastable transaction to raw bytes
```
Comment (catenocrypt) [inline include/TrustWalletCore/TWAnySigner.h:26]: As both input&output are Data type, their type does not tell much, comment is important.  What is 'raw bytes'?  I think a bit more specific description is needed here.

File: include/TrustWalletCore/TWAnySigner.h [hunk 2]
```diff
@@ -23,6 +23,12 @@ extern TWString *_Nonnull TWAnySignerSignJSON(TWString *_Nonnull json, TWData *_
...
+extern TWData *_Nonnull TWAnySignerEncode(TWData *_Nonnull input, enum TWCoinType coin);
+
+/// Decodes raw transaction bytes to representable format
```
Comment (catenocrypt) [inline include/TrustWalletCore/TWAnySigner.h:29]: Let's be consistent, 'representable format' vs. 'broadcastable transaction'

File: src/Ethereum/RLP.cpp [hunk 3]
```diff
@@ -134,3 +140,127 @@ Data RLP::putint(uint64_t i) noexcept {
...
+    }
+    RLP::DecodedItem item;
+    auto inputLen = input.size();
+    auto prefix = input[0];
+    if (prefix <= 0x7f) {
+        // a single byte whose value is in the [0x00, 0x7f] range, that byte is its own RLP encoding.
+        item.decoded.push_back(Data{input[0]});
+        item.remainder = Data(input.begin() + 1, input.end());
+        return item;
+    } else if (prefix <= 0xb7) {
```
Comment (catenocrypt) [inline src/Ethereum/RLP.cpp:177]: All if branches end with a return; else is unnecessary.

Comment (hewigovens) [reply src/Ethereum/RLP.cpp]: thanks, all done

**Comment by catenocrypt** [comment]
Ready for review?

**Comment by hewigovens** [comment]
> Ready for review?

++

**Pull request merged**